modules/lib/lib-content/src/main/resources/lib/xp/content.ts

```diff
@@ -18,6 +18,29 @@ declare global {
 
 import type {ByteSource, Content, PublishInfo} from '@enonic-types/core';
 
+const isString = (value: unknown): value is string => value instanceof String || typeof value === 'string';
+
+const isNumber = (value: unknown): value is number => typeof value === 'number' && isFinite(value);
+
+function checkRequiredString<T extends object>(obj: T, name: keyof T): void {
+    checkRequired(obj, name);
+    if (!isString(obj[name])) {
+        throw `Required parameter '${String(name)}' is not a string!`;
+    }
+}
+
+function checkOptionalString<T extends object>(obj: T, name: keyof T): void {
+    if (obj != null && obj[name] != null && !isString(obj[name])) {
```

Comment:
```ts
obj != null && obj[name] != null
```
can be simplified to `obj?.[name] != null`

Reply:
Yeah, I have not started using ?. I'm uncertain wether all build tools support it...

Reply:
But if the build tool used in XP supports it, that's not an issue here.

Reply:
@ComLock Oh, the TS compiler certainly do. Although, with our target (ES5) it will be transform using ternary operator:
```js
(obj === null || obj === void 0 ? void 0 : obj[name]) != null && obj[name]
```
Nevertheless, it will work as expected both in runtime and for types. So, feel free to use it in modern TS if it'll make your code better.

Reply:
If it works with webpack/babel and esbuild/swc then I probably will start using it.